app/packages/looker/src/overlays/util.ts

```diff
@@ -132,3 +140,116 @@ export const getHashLabel = (label: RegularLabel): string => {
     return `${label.label}.${label.id}`;
   }
 };
+
+export const shouldShowLabelTag = (
+  selectedLabelTags: string[], // labelTags that are active
+  labelTags: string[] // current label's tags
+) => {
+  return (
+    (selectedLabelTags?.length == 0 && labelTags.length > 0) ||
+    selectedLabelTags?.some((tag) => labelTags.includes(tag))
+  );
+};
+
+type LabelColorProps = {
+  coloring: Coloring;
+  path: string;
+  label: RegularLabel;
+  isTagged: boolean;
+  labelTagColors: CustomizeColor;
+  customizeColorSetting: CustomizeColor[];
+};
+
+export const getLabelColor = ({
```

Comment:
Awesome to see this as a new function. Is there an opportunity to break out the main if statements into helper functions? Keeping if statement nesting to a minimum helps readability

Reply:
Yes, just broke the function up